src/genie/libs/parser/iosxe/show_ap.py

```diff
@@ -295,3 +295,782 @@ def cli(self, output=None):
+        chan_energy_average_capture = re.compile(r"^\s+Average\s+:\s+(?P<chan_energy_average_dbm>\S+)\s+dBm")
+        #     Maximum                                  : -81 dBm
+        chan_energy_max_capture = re.compile(r"^\s+Maximum\s+:\s+(?P<chan_energy_max_dbm>\S+)\s+dBm")
+        #   Channel Dwell Times
+        chan_dwell_times_capture = re.compile(r"^\s+Channel\s+Dwell\s+Times$")
+        #     Minimum                                  : 4 hours 9 minutes 54 seconds
+        chan_dwell_minimum_capture = re.compile(
+            r"^\s+Minimum\s+:\s+(?P<chan_dwell_min_hours>\d+)\s+hours\s+(?P<chan_dwell_min_minutes>\d+)\s+minutes\s+(?P<chan_dwell_min_seconds>\d+)\s+seconds$")
+        #     Average                                  : 4 hours 24 minutes 54 seconds
+        chan_dwell_average_capture = re.compile(
+            r"^\s+Average\s+:\s+(?P<chan_dwell_average_hours>\d+)\s+hours\s+(?P<chan_dwell_average_minutes>\d+)\s+minutes\s+(?P<chan_dwell_average_second>\d+)\s+seconds$")
+        #     Maximum                                  : 4 hours 26 minutes 35 seconds
+        chan_dwell_max_capture = re.compile(
+            r"^\s+Maximum\s+:\s+(?P<chan_dwell_max_hours>\d+)\s+hours\s+(?P<chan_dwell_max_minutes>\d+)\s+minutes\s+(?P<chan_dwell_max_seconds>\d+)\s+seconds$")
+        #   802.11a 5 GHz Auto-RF Channel List
+        channel_list_capture = re.compile(r"^\s+802.11a\s+5\s+GHz\s+Auto-RF\s+Channel\s+List$")
+        #     Allowed Channel List                     : 36,40,44,48,52,56,60,64,100,104,108,112,116,120,124,128,132,136,140,144,149,153,157,161
+        allowed_channel_list_capture = re.compile(r"^\s+Allowed\s+Channel\s+List\s+:\s+(?P<allowed_channel_list>\S+)$")
+        #     Unused Channel List                      : 165
+        unused_channel_list_capture = re.compile(r"^\s+Unused\s+Channel\s+List\s+:\s+(?P<unused_channel_list>\d+)$")
+
+
+        def change_data_type(value):
+            if value.isdigit():
+                value = value.strip()
+                value = int(value)
+            else:
+                try:
+                    # Change strings to float if possible
+                    value = float(value)
+                except ValueError:
+                    # if the value is not an int or float, leave it as a string.
+                    pass
+            return value
+
+        for line in out:
+            # Leader Automatic Channel Assignment
+            if lead_auto_chan_assn_capture.match(line):
+                lead_auto_chan_assn_capture_match = lead_auto_chan_assn_capture.match(line)
+                groups = lead_auto_chan_assn_capture_match.groupdict()
+                if not show_ap_dot11_5ghz_channel_dict.get('channel_assignment', {}):
+                    show_ap_dot11_5ghz_channel_dict['channel_assignment'] = {}
+                continue
+            #   Channel Assignment Mode                    : AUTO
+            elif chan_assn_mode_capture.match(line):
```

Comment:
No need for using `elif` since you are issuing `continue` after matching each pattern